sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/RequestContext.java

```diff
@@ -167,16 +163,15 @@ public ClientLogger getLogger() {
      * otherwise a new header will be created.</p>
      *
      * @param header The header key.
-     * @return The updated {@link RequestOptions} object.
+     * @return The updated {@link RequestContext} object.
```

Comment:
The JavaDoc for this type, if we go ahead with this pattern, will need updating to state that it is a new instance rather than an updated version of the same instance.